x-pack/plugins/cases/public/components/user_actions/index.test.tsx

```diff
@@ -75,143 +75,158 @@ const useFindCaseUserActionsMock = useFindCaseUserActions as jest.Mock;
+        ...defaultUseFindCaseUserActions,
+        data: { userActions: ourActions },
+      });
 
-    appMockRender.render(<UserActions {...defaultProps} />);
+      appMockRender.render(<UserActions {...defaultProps} />);
 
-    userEvent.click(
-      within(
-        screen.getAllByTestId(`comment-create-action-${defaultProps.data.comments[0].id}`)[1]
-      ).getByTestId('property-actions-user-action-ellipses')
-    );
+      userEvent.click(
+        await within(
+          (
+            await screen.findAllByTestId(
+              `comment-create-action-${defaultProps.data.comments[0].id}`
+            )
+          )[1]
+        ).findByTestId('property-actions-user-action-ellipses')
+      );
 
-    await waitForEuiPopoverOpen();
+      await waitForEuiPopoverOpen();
 
-    userEvent.click(screen.getByTestId('property-actions-user-action-pencil'));
+      userEvent.click(await screen.findByTestId('property-actions-user-action-pencil'));
 
-    userEvent.click(
-      within(
-        screen.getAllByTestId(`comment-create-action-${defaultProps.data.comments[0].id}`)[1]
-      ).getByTestId('editable-cancel-markdown')
-    );
+      userEvent.click(
+        await within(
+          (
+            await screen.findAllByTestId(
+              `comment-create-action-${defaultProps.data.comments[0].id}`
+            )
+          )[1]
+        ).findByTestId('editable-cancel-markdown')
+      );
 
-    await waitFor(() => {
       expect(
         within(
-          screen.getAllByTestId(`comment-create-action-${defaultProps.data.comments[0].id}`)[1]
+          (
+            await screen.findAllByTestId(
+              `comment-create-action-${defaultProps.data.comments[0].id}`
+            )
+          )[1]
         ).queryByTestId('editable-markdown-form')
       ).not.toBeInTheDocument();
     });
-  });
 
-  it('calls update comment when comment markdown is saved', async () => {
-    const ourActions = [getUserAction('comment', UserActionActions.create)];
+    it('calls update comment when comment markdown is saved', async () => {
+      const ourActions = [getUserAction('comment', UserActionActions.create)];
 
-    useFindCaseUserActionsMock.mockReturnValue({
-      ...defaultUseFindCaseUserActions,
-      data: { userActions: ourActions },
-    });
+      useFindCaseUserActionsMock.mockReturnValue({
+        ...defaultUseFindCaseUserActions,
+        data: { userActions: ourActions },
+      });
 
-    appMockRender.render(<UserActions {...defaultProps} />);
+      appMockRender.render(<UserActions {...defaultProps} />);
 
-    userEvent.click(
-      within(
-        screen.getAllByTestId(`comment-create-action-${defaultProps.data.comments[0].id}`)[1]
-      ).getByTestId('property-actions-user-action-ellipses')
-    );
+      userEvent.click(
+        await within(
+          (
+            await screen.findAllByTestId(
+              `comment-create-action-${defaultProps.data.comments[0].id}`
+            )
+          )[1]
```

Comment:
Assuming there is only one `create` action and we even have the comment id, why is this `[1]` ?

Reply:
hmm.. not sure but it shows multiple elements with same id.

Reply:
I think I got it, it's because it has 2 user actions list and both have same mock used. So there are two comment create action with same id. 😃 

Good catch!!

Since this test focuses only on markdown behaviour, no need of 2 user actions lists. I will remove additional list where not needed.